src/main/java/rest/ApplicationClient.java

```diff
@@ -0,0 +1,96 @@
+        String responseBody = ApiUtils.fromResponseToString(response, HttpStatus.SC_OK).replace("[", "")
+                .replace("]", "").trim();
+        return ApiUtils.getUserFromJson(responseBody);
+    }
+
+    public static List<User> getFilterUsers(int age, String parameter) {
+        CloseableHttpResponse response = CloseableHttpClient.sendGet(applicationData.getProperty("api.users"), parameter, age);
+        String responseBody = ApiUtils.fromResponseToString(response, HttpStatus.SC_OK).replace("[", "")
+                .replace("]", "").trim();
+        return ApiUtils.getUserFromJson(responseBody);
+    }
+
+    public static List<User> getFilterUsers(String sex, String parameter) {
+        CloseableHttpResponse response = CloseableHttpClient.sendGet(applicationData.getProperty("api.users"), parameter, sex);
+        String responseBody = ApiUtils.fromResponseToString(response, HttpStatus.SC_OK).replace("[", "")
+                .replace("]", "").trim();
+        return ApiUtils.getUserFromJson(responseBody);
+    }
+
+    public static void updateUser(User oldUser, User newUser) {
+        String json = "{" + "\"userNewValues\"" + ":" + ApiUtils.fromObjectToJson(newUser)
+                + "," + "\"userToChange\"" + ":" + ApiUtils.fromObjectToJson(oldUser) + "}";
+        CloseableHttpResponse response = CloseableHttpClient.sendPut(applicationData.getProperty("api.users"), json);
+        ApiUtils.fromResponseToString(response, HttpStatus.SC_OK);
+    }
+
+    public static void updateUser(User oldUser, User newUser, int httpStatus) {
+        String json = "{" + "\"userNewValues\"" + ":" + ApiUtils.fromObjectToJson(newUser)
+                + "," + "\"userToChange\"" + ":" + ApiUtils.fromObjectToJson(oldUser) + "}";
+        CloseableHttpResponse response = CloseableHttpClient.sendPut(applicationData.getProperty("api.users"), json);
+        ApiUtils.fromResponseToString(response, httpStatus);
+    }
+
+    public static void deleteUser(String json) {
+        CloseableHttpResponse response = CloseableHttpClient.sendDelete(applicationData.getProperty("api.users"), json);
+        String responseBody = ApiUtils.fromResponseToString(response, HttpStatus.SC_NO_CONTENT);
+        ApiUtils.getUserFromJson(responseBody);
+    }
+
+    public static void deleteUser(String json, int httpStatus) {
+        CloseableHttpResponse response = CloseableHttpClient.sendDelete(applicationData.getProperty("api.users"), json);
+        String responseBody = ApiUtils.fromResponseToString(response, httpStatus);
+        ApiUtils.getUserFromJson(responseBody);
+    }
+
+    public static String uploadUser(File file) {
+        CloseableHttpResponse response = CloseableHttpClient.sendPost(applicationData.getProperty("api.users.upload"), file);
+        return ApiUtils.fromResponseToString(response, HttpStatus.SC_CREATED).replace("Number of users = ", "").trim();
+    }
```

Comment:
method should reuse uploadUser(File file, int httpStatus)

Reply:
fixed